### PR TITLE
Merge cloudflare branch into master

### PR DESCRIPTION
**Pull request by marmeladema**

### CODE REVIEW

File: engine/benches/bench.rs [hunk 1]
```diff
@@ -242,6 +244,41 @@ fn bench_string_function_comparison(c: &mut Criterion) {
...
+    ];
```
Comment (inikulin) [inline engine/benches/bench.rs:250]: I doubt it's a good idea to have this in the public code

Comment (marmeladema) [reply engine/benches/bench.rs:250]: Sure, i'll provide a PR quickly to change that.

**Comment by marmeladema** [comment]
@inikulin would it be possible to merge this PR with a merge commit? I'd like to avoid rewriting the history of that branch if possible.

File: engine/benches/bench.rs (hunk 1 above)
Comment (RReverser) [reply engine/benches/bench.rs:250]: You might want to remove it from history altogether before merging if it's a sensitive information.

Comment (marmeladema) [reply engine/benches/bench.rs:250]: We have other projects pointing at commits from this branch so rewriting history will forbid us to build older versions of our internal tools unless we keep/fork this branch internally. I don't believe its sensitive information but i'll check. Also, that's unfortunate but other people might have already checked out / forked this branch so I am afraid we won't be able to remove it from the internet anyway :-(

**Comment by inikulin** [comment]
@marmeladema I'll reply in a private chat

**Pull request closed without merging**